relation/src/gadgets/logic.rs

```diff
@@ -192,6 +192,47 @@ impl<F: PrimeField> PlonkCircuit<F> {
         self.insert_gate(&wire_vars, Box::new(CondSelectGate))?;
         Ok(y)
     }
+
+    /// Constrain a variable `y` to equal `f_0` if `b` is zero, or `f_1` if `b` is
+    /// one. Return error if variables are invalid.
+    pub fn const_conditional_select_gate(
+        &mut self,
+        b: BoolVar,
+        y: Variable,
+        f_0: F,
+        f_1: F,
+    ) -> Result<Variable, CircuitError> {
+        self.check_var_bound(b.into())?;
+        self.check_var_bound(y)?;
+
+        let wire_vars = [b.into(), self.zero(), self.zero(), self.zero(), y];
+        self.insert_gate(&wire_vars, Box::new(ConstCondSelectGate { f_0, f_1 }))?;
+        Ok(y)
+    }
+
+    /// Obtain a variable that equals `f_0` if `b` is zero, or `f_1` if `b` is
+    /// one. Return error if variables are invalid.
+    pub fn const_conditional_select(
+        &mut self,
+        b: BoolVar,
+        f_0: F,
+        f_1: F,
+    ) -> Result<Variable, CircuitError> {
+        self.check_var_bound(b.into())?;
+
+        // y = x_bit
+        let y = if self.witness(b.into())? == F::zero() {
+            self.create_variable(f_0)?
+        } else if self.witness(b.into())? == F::one() {
+            self.create_variable(f_1)?
+        } else {
+            return Err(CircuitError::ParameterError(
+                "b in Constant Conditional Selection gate is not a boolean variable".to_string(),
+            ));
+        };
```

Comment:
let bw = self.witness(b.into())?;
if bw != F::zero() && bw != F::one() {
    return Err(CircuitError::ParameterError(
        "b in Constant Conditional Selection gate is not boolean".to_string(),
    ));
}